### PR TITLE
Add edit/delete flow to bot messages

### PR DESCRIPTION
**Pull request by alexquick**
This adds a Slack shortcut to edit/delete bot messages. I still want to test this a bit more but it does what it says on the tin.

Right now it restricts editors based on one of two criteria:
 - You can edit a post if you are mentioned in it.
 - You can edit a post if you are a member of #admins or #intake_volunteers. This will necessitate inviting the bot to those channels.

edit_request

closes: #41

### CODE REVIEW

File: src/slackapp/lib/channels.js [hunk 1]
```diff
@@ -30,7 +33,7 @@ module.exports.findChannelByName = async name => {
 };
 
 /**
- * Finds a channel by its #name. Returns undefined if no element matches
```
Comment (mjmaurer) [inline src/slackapp/lib/channels.js:33]: Thanks :)

File: src/slackapp/flows/editPost.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  return successResponse("The message was successfully updated");
+}
+
+async function deletePost(payload) {
+  const meta = JSON.parse(payload.view.private_metadata);
+  try {
+    await slackapi.chat.delete({
+      channel: meta.message_channel,
+      ts: meta.message_ts
+    });
```
Comment (mjmaurer) [inline src/slackapp/flows/editPost.js:98]: It'd be cool but probably out of scope to clear the slack message data for a request so the bot could post again. would you mind filing a PR for this?

Comment (mjmaurer) [reply src/slackapp/flows/editPost.js:98]: Airtable formulas support string operations so we could add findInMeta to the api or something. I'll leave it up to you if you want to include in a separate PR

**Comment by alexquick** [comment]
Thanks for the review-- no worries about the delay, I've been busy breaking stuff at work anyhow.

Gating by channel membership was @mab253 's awesome idea. Slack mention-groups are really lame so I'm happy we all agree this is the way to go.

I'll make it delete clear the old message metadata tonight or tomorrow and we can go from there.

**Comment by mjmaurer** [comment]
Great PR. I meant issue not PR for the followup :/

**Pull request merged**